left-sidebar: Hide the unread counters on hover

somesh202 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
**UI Experiment**
Description:
- Hide the unread counters on hover for any lines that have an associated three-dot menu.
- Have the three-dot menus appear in the location of the unread counters.
- Do not show the three-dot stream menu when hovering over a topic within that stream.
- Shift the Streams + menu to be aligned with the unread message counters.

Fixes #20935

**Testing plan:** <!-- How have you tested? -->
Manually
**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
![Zulip-4](https://user-images.githubusercontent.com/77766761/154830423-b9d6c433-2173-488e-9536-0703b252598e.gif)




<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

somesh202 commented:
@alya Can you have a look at this?

alya commented:
Sorry for the delay on this one! Playing around with it, it's definitely a bit too glitchy to deploy at this point.

1. You sometimes get the three-dot menu superimposed on the unread counter. I see this consistently with All topics, and sometimes (not sure when) for streams:
![Screen Shot 2022-03-02 at 11 37 55 PM](https://user-images.githubusercontent.com/2090066/156518835-cbedc235-6ba1-4980-95d4-02bd4206b869.png)

![Screen Shot 2022-03-02 at 11 38 27 PM](https://user-images.githubusercontent.com/2090066/156518826-caf0a6d7-5d66-413a-a3b5-69a734761344.png)

2. The three-dot topic menu has different horizontal alignment for topics vs. streams; it should be the same.
3. Long topic names go further right than they should:

![Screen Shot 2022-03-02 at 11 39 36 PM](https://user-images.githubusercontent.com/2090066/156519242-505ba8b2-1d5d-4623-a2a1-d098f076c658.png)

alya commented:
By the way, we have recently updated our guide on reviewing code (including your own), and I highly recommend going through the [manual review section](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#manual-testing) when you believe the feedback above has been addressed. Thanks!

somesh202 commented:
@alya  Thanks for your feedback! I've pushed some changes addressing your feedback to this PR and Please take a look.

- Although, I've fixed the extension of topic-names on hover. But, as we are changing the position of the three-dot-menu to be aligned with the unread-counters, they get overlapped with the topics, when we have a long-topic name.  So, I've shifted the unread counters a bit right by some pixels, which would avoid that problem. 

![Zulip-UI-20935](https://user-images.githubusercontent.com/77766761/156726858-8074af55-b76d-4db2-a897-d030a66b2ee5.gif)

alya commented:
@somesh202 As noted above, please address:

> The three-dot topic menu has different horizontal alignment for topics vs. streams; it should be the same.

![Screen Shot 2022-03-07 at 11 10 49 PM](https://user-images.githubusercontent.com/2090066/157185390-3215ddb4-692f-4339-ac67-40f67db04b5b.png)

The stream counter alignment looks approximately correct.

Thanks!

alya commented:
Please post a screenshot demonstrating that this has been fixed when you've had a chance to take care of it.

somesh202 commented:
@alya Here are the screenshots after fixing the above.

![Screenshot 2022-03-09 022726](https://user-images.githubusercontent.com/77766761/157374212-095f89ec-3bdf-47c3-984a-e9b74c1d578e.png)
![Screenshot 2022-03-09 022754](https://user-images.githubusercontent.com/77766761/157374215-a3e02168-00f6-4a9c-84ae-0d6acdcb4281.png)

alya commented:
Thanks! I just noticed that the alignment of the three dots is different in the top section vs. the STREAMS section, which needs to be fixed as well.

somesh202 commented:
![Screenshot 2022-03-09 180209](https://user-images.githubusercontent.com/77766761/157442495-1e3e82c9-f810-4e7b-802c-756267d512b5.png)
@alya Here's the screenshot after fixing the alignment of three-dot menu.

alya commented:
Can you post screenshot(s) or a screen capture that demonstrates that the alignment has been fixed? It's impossible to tell from the one above. Thanks!

somesh202 commented:
@alya  Sorry,  
Here are the screenshots and screencapture for the above fix. 
![Zulip-20935](https://user-images.githubusercontent.com/77766761/157612382-698fe159-bc72-4b9f-9ea0-a01250a75bd8.gif)
![Screenshot 2022-03-10 125343](https://user-images.githubusercontent.com/77766761/157612046-5d0e040a-8df0-468b-b169-d2cd2b7bc1d2.png)
![Screenshot 2022-03-10 125410](https://user-images.githubusercontent.com/77766761/157612055-4695ff20-3ea6-4c4b-b4d7-9ab4358561af.png)

somesh202 commented:
@alya Could you please re-open this PR? It was closed accidentally.

timabbott commented:
Closing since as discussed in https://github.com/zulip/zulip/issues/20935, we decided on a different UI direction after seeing this in action; thanks so much for prototyping this @somesh202!
